### DISCUSSION TITLE
Actions disappear on resource details page

### DISCUSSION THREAD

**Discussion by timcv** [laravel/nova-issues#4870] [Ideas & Feature Requests]
- Laravel Version: 9.31.0
- Nova Version: 4.15.1
- PHP Version: 8.1
- Operating System and Version: MacOS 12.3
- Browser type and version: Chrome 104.0.5112.79

### Description:
When I place a HasMany field at the top of a resource details page the action buttons disappear.

HasMany::make() last in the fields() response array:
![Skärmavbild 2022-09-27 kl  13 08 52](https://user-images.githubusercontent.com/1415514/192517630-d0c334a7-8cd3-496d-8670-9c74db87601c.png)

HasMany::make() first in the fields() response array:
![Skärmavbild 2022-09-27 kl  13 08 32](https://user-images.githubusercontent.com/1415514/192517644-cd4bd9f2-8c9c-43ea-8339-2d43979efb9a.png)

### Detailed steps to reproduce the issue on a fresh Nova installation:
Put a HasMany field first in the fields() function in a resource.

**Comment by timcv** [comment]
@crynobone Do you need any more info from me? This bug is stalling my update to Nova 4.

Im really looking forward to put all new nova 4 goodies in production.

Kind regards
/Tim

**Comment by crynobone** [comment]
The current behaviour is expected as plan for Nova 4.

Moving the toolbar down give a bad UX in my opinion and any changes would depends on whether the team agree to change the current behaviour. Haven't received any yet and currently not specifying this as a bug.

**Reply by timcv** [reply]
Hi, have you heard anythng about this issue? In my opinion it feels very strange that the possibility to run actions from the detail page is denyed when there is a HasMany or a custom field above the default field group.

/Tim

**Reply by crynobone** [reply]
No plan to change this at the moment.

It's also weird to have `Images sources` at the top of `Site detail` when the page should be focusing on `Site detail`.

**Reply by timcv** [reply]
Aha, that was just an example to reproduce the bug. I dont use it for that case. I use Nova to build a CRM system handeling sales and orders.

One example of my use case is a resource tool at the top witch allows the user to quickly step between different resources:
![Skärmavbild 2022-10-12 kl  10 35 18](https://user-images.githubusercontent.com/1415514/195293325-a9c69bda-6efd-45fd-a848-8a36f848bb3a.png)

Another of my use cases is to handle a flow for reviwing changes to the resources where i store the change data in another table while waiting for an admin to review the changes.
![Skärmavbild 2022-10-12 kl  10 44 07](https://user-images.githubusercontent.com/1415514/195295355-b2a1e297-298d-4526-a380-954e20cb3500.png)

A third usecase is that use the main level resource as a cointainer for the sub resources where the HasMany list resources is where the main focus is put. But i still want to be able to use my action for generating a PDF.

I simple option to decide this by my self if i want to turn on/off the edit/actions button would be very important for the system i worked almost fulltime on the last 3 years. I completely love the rest of the Nova4 updates.

Kind regards
Tim

**Reply by crynobone** [reply]
To archive that level of customisable you should look into creating custom Tool, as for Resource Tool that can be replaced with Resource Detail Card (you can create Card and not just Metric).

**Comment by flowan** [comment]
This also happens when using a custom Resource Tool. Is there a solution?

**Comment by atmikev** [comment]
In my opinion this is either a bug or a poor UX decision. I debugged for an hour thinking there was an issue on my side. Its definitely not obvious that the resource buttons just disappear if you add certain components. It should be a custom setting developers can control.
